engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/details/repository-details.controller.js

```diff
@@ -30,6 +30,33 @@
             $scope.page.loading = false;
         }
 
+        $scope.repositoryVersions = function () {
+            var publishedViews = {};
+            angular.forEach($scope.repository.versions, function (version) {
+                if (publishedViews[version.cv_id]) {
```

Comment:
Could a `reduce` method eliminate the need to two nearly identical  if/else blocks?

Reply:
Could be a mental block but would still need to check if key exists to push to existing array vs initializing an array to add.

```
  var versionHash = {
        versionId: version.id,
        cvId: version.cv_id,
        cvName: version.cv_name,
        version: version.version
  };

  if (publishedViews[version.cv_id]) {
      publishedViews[version.cv_id].push(versionHash);
  } else {
      publishedViews[version.cv_id] = [versionHash];
  }
```
Does this make it better?

Reply:
Looks like underscore.js gives us a groupBy!

```
        $scope.repositoryVersions = function () {                                                                                                                                                                                             
          return _.groupBy($scope.repository.versions, function(version) {                                                                                                                                                                    
            return version.cv_id;                                                                                                                                                                                                             
          });                                                                                                                                                                                                                                 
        }; 
```

I'm not sure why you needed to rename the keys of the API response for versions